api/python/provisioner/commands/check.py

```diff
@@ -609,6 +616,177 @@ def _cluster_status(*, args: str) -> CheckEntry:
+            err_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                             comment="Package cortx-py-utils not installed")
+            res.append(err_res)
+            return res
+
+        check_value : CheckEntry = CheckEntry(cfg.Checks.NETWORK_HCA.value)
+        try:
+            nodes = Check._get_pillar_data("cluster/node_list")
+            check_res : CheckEntry = CheckEntry(cfg.Checks.NETWORK_HCA.value)
+            try:
+                hca_args = provider + nodes
+                NetworkV().validate('hca', hca_args)
+                check_res.set_passed(checked_target=cfg.ALL_MINIONS,
+                                     comment="HCA Presence and "
+                                     f"Ports Validated: {hca_args}")
+
+            except Exception as exc:
+                check_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                                   comment=str(exc))
+            res.append(check_res)
+
+        except Exception as exc:
+            check_value.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            res.append(check_value)
+
+        return res
+
+    @staticmethod
+    def _storage_luns(*, args: str) -> List[CheckEntry]:
+        """Check storage LUNs
+        storage_luns <v_check> <nodes>
+        """
+
+        res: List[CheckEntry] = list()
+
+        if cortx_py_utils_import_error:
+            err_res : CheckEntry = CheckEntry(cfg.Checks.STORAGE_LUNS.value)
+            err_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                             comment="Package cortx-py-utils not installed")
+            res.append(err_res)
+            return res
+
+        check_value : CheckEntry = CheckEntry(cfg.Checks.STORAGE_LUNS.value)
+        try:
+            nodes = Check._get_pillar_data("cluster/node_list")
+            available_luns_checks = (cfg.LUNS_CHECKS)
+
+            for luns_check in available_luns_checks:
+                check_res : CheckEntry = CheckEntry(
+                                         cfg.Checks.STORAGE_LUNS.value)
+                try:
+                    lun_args = [luns_check] + nodes
+                    StorageV().validate('luns', lun_args)
+                    check_res.set_passed(checked_target=cfg.ALL_MINIONS,
+                                         comment="LUNs Accessibility, " 
+                                         "Volume Size and Mapping "
+                                         f"Validated: {lun_args}")
+
+                except Exception as exc:
+                    check_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                                       comment=str(exc))
+                res.append(check_res)
+        except Exception as exc:
+            check_value.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            res.append(check_value)
+
+        return res
```

Comment:
The same remarks as for `_network_drivers` method. This code also can be optimized and simplified.

Reply:
Comment addressed.

Reply:
The same as for `_network_hca`.
It can be just simplified as 
```python
@staticmethod
def _storage_luns(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
    """
    Check if Storage LUNs are proper
    :param args: storage_luns check specific parameters and arguments
    :return:
    """
    check_value: CheckEntry = CheckEntry(cfg.Checks.STORAGE_LUNS.value)
    if cortx_py_utils_import_error:
        check_value.set_fail(checked_target=cfg.ALL_MINIONS,
                             comment="Package cortx-py-utils not installed")
        return check_value

    try:
        nodes = Check._get_pillar_data("cluster/node_list")
    except Exception as exc:
        check_value.set_fail(checked_target=cfg.ALL_MINIONS,
                             comment=str(exc))
        return check_value

    res: List[CheckEntry] = list()
    for luns_check in cfg.LUNS_CHECKS:
        check_res: CheckEntry = CheckEntry(cfg.Checks.STORAGE_LUNS.value)
        try:
            lun_args = [luns_check] + nodes
            StorageV().validate('luns', lun_args)
        except Exception as exc:
            check_res.set_fail(checked_target=cfg.ALL_MINIONS,
                               comment=str(exc))
        else:
            check_res.set_passed(checked_target=cfg.ALL_MINIONS,
                                 comment="LUNs Accessibility, "
                                         "Volume Size and Mapping "
                                         f"Validated: {lun_args}")
        res.append(check_res)

    return res
```

Reply:
Comment addressed.